src/dune_engine/dep.ml

```diff
@@ -63,9 +63,7 @@ module T = struct
     | Alias a -> Some (File (trace_file (Path.build (Alias.stamp_file a))))
     | File_selector dir_glob ->
       let id = File_selector.to_dyn dir_glob
-      and files =
-        eval_pred dir_glob |> Path.Set.to_list |> List.map ~f:trace_file
-      in
+      and files = eval_pred dir_glob |> Path.Set.to_list_map ~f:trace_file in
```

Comment:
@aalekseyev this bit is a hot spot. Shall we just write an explicit fold here perhaps?

Reply:
I don't mind the existence of `to_list_map_rev` if that's your preferred solution, I just think it should be used sparingly. 
Seems fine to use it here.
A fold written inline seems fine too.